### DISCUSSION TITLE
How to design two beam sections with different radius, mass density, and young's modulus?

### DISCUSSION THREAD

**Discussion by Junang-Wang** [sofa-framework/sofa#4446] [BeamAdapter] [answered]
Hi,
I am studying a beam that involves two sections with different radii, mass densities, and Young's moduli. However, I found that `AdaptiveBeamForceFieldAndMass` and `WireBeamInterpolation` accept a single radius and mass density. Does anyone know how to solve this problem? Or do I need to consider the two sections as two individual beams and then join them into one beam?

**Comment by epernod** [comment, marked as answer]
Hi @Junang-Wang ,
What you want is work in progress. We are going to move the radius and mass density in the `RodSection `component to allow a much finer granularity modelling. 
In the mean time maybe you could use 2 different tools combined together. check the example: https://github.com/sofa-framework/BeamAdapter/blob/master/examples/3instruments_collis.scn
where we have several tools and therefor several `AdaptiveBeamForceFieldAndMass `and `WireBeamInterpolation`

**Reply by Junang-Wang** [reply]
Thanks @epernod,
Looking forward to your work
